### PR TITLE
fix: updated the design for the admin users

### LINKED ISSUE #29692: [Task]: For self-hosted admin users, we need to update designs for setup form
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- Background image to make it consistent with the non-admin users signup success screen.
- Profiling questions to be similar in design with the non-admin signup success screen.

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
This PR updates the design of the admin users UI for profiling questions to newer version. Below are the list of changes introduced in this PR : 
- Deprecated `role` property for the user
- Updated components to reuse the similar UI
- Updated the background image for the admin setup screen
  - before Screenshot 2023-12-21 at 10 29 52 PM

  - after Screenshot 2023-12-21 at 10 51 58 PM

- Changed the profiling questions for the admin second page while setting up the instance
  - before Screenshot 2023-12-21 at 10 30 16 PM

  - after Screenshot 2023-12-21 at 10 51 48 PM

#### PR fixes following issue(s)
Fixes #29692
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `UserWelcomeScreen` component to enhance the user onboarding experience.
  - Added a proficiency level selection to the user setup process.
  - Updated the use case selection options for a more tailored setup.

- **Bug Fixes**
  - Fixed issues with form input handling for proficiency and use case selections.

- **Refactor**
  - Streamlined the setup process by removing role selection and custom use case input.
  - Refined the user interface elements related to the setup forms.

- **Documentation**
  - Updated constant messages to align with the new setup flow.

- **Style**
  - Implemented new styles for the `WelcomeBackground` and its components to improve visual appeal.

### CODE REVIEW

File: app/client/src/pages/setup/DetailsForm.tsx [hunk 1]
```diff
@@ -1,42 +1,38 @@
...
+import { Space } from "./NonSuperUserProfilingQuestions";
 
 export interface DetailsFormValues {
   firstName?: string;
   lastName?: string;
   email?: string;
   password?: string;
   verifyPassword?: string;
-  role?: string;
+  proficiency?: string;
```
Comment (sneha122) [inline app/client/src/pages/setup/DetailsForm.tsx:34]: I see that we are storing the proficiency information, do we not want to store use case as well?

**Comment by sneha122** [comment]
@AmanAgarwal041 Tested the PR for following cases:
- Signed up as admin user -> I can see the profiling questions after filling in the details, I can also see the checkbox with "I want updates" -> Working as expected
- Signed up as non admin user -> I can see profiling questions as soon as I sign up, I do not see the checkbox of "I want updates" -> Working as expected

File: app/client/src/pages/setup/DetailsForm.tsx (hunk 1 above)
Comment (AmanAgarwal041) [reply app/client/src/pages/setup/DetailsForm.tsx:34]: It has a property `useCase` in the same type : https://github.com/appsmithorg/appsmith/pull/29712/files#diff-27b6d74fb1347fac38b104faba7e52b26e8adababc8e9c9f57506ca01aaa7b5fR35

**Review by sneha122 — DISMISSED** [review]

**Comment by sneha122** [comment]
@AmanAgarwal041 Tested the EE dp as well, works as expected there too

**Review by sneha122 — APPROVED** [review]

**Pull request merged**